whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/ElasticAutocompleteSearch.java

```diff
@@ -0,0 +1,112 @@
+package net.ripe.db.whois.api.autocomplete;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import net.ripe.db.whois.common.elasticsearch.ElasticIndexService;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchType;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.MultiMatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.TermsQueryBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.SortBuilders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Component
+@Conditional(ElasticSearchCondition.class)
```

Comment:
Could you use ConditionalOnProperty instead? You can then drop the Conditional classes, may be simpler (less code)

```
@ConditionalOnProperty(prefix = "elasticsearch", name = "enabled", havingValue = "true")
```

Reply:
ConditionalOnProperty is springBoot annotation in spring we have only Conditional